resources/rebuild.sh

```diff
@@ -199,31 +201,113 @@ function build_al_kernel {
     popd &>/dev/null
 }
 
-#### main ####
+function prepare_and_build_rootfs {
+    BIN=overlay/usr/local/bin
+    compile_and_install $BIN/init.c $BIN/init
+    compile_and_install $BIN/fillmem.c $BIN/fillmem
+    compile_and_install $BIN/fast_page_fault_helper.c $BIN/fast_page_fault_helper
+    compile_and_install $BIN/readmem.c $BIN/readmem
+    if [ $ARCH == "aarch64" ]; then
+        compile_and_install $BIN/devmemread.c $BIN/devmemread
+    fi
 
-install_dependencies
+    build_rootfs ubuntu-22.04 jammy
+    build_initramfs
+}
 
-BIN=overlay/usr/local/bin
-compile_and_install $BIN/init.c    $BIN/init
-compile_and_install $BIN/fillmem.c $BIN/fillmem
-compile_and_install $BIN/fast_page_fault_helper.c $BIN/fast_page_fault_helper
-compile_and_install $BIN/readmem.c $BIN/readmem
-if [ $ARCH == "aarch64" ]; then
-    compile_and_install $BIN/devmemread.c $BIN/devmemread
-fi
+function build_al_kernels {
+    if [[ $# = 0 ]]; then
+        local KERNEL_VERSION="all"
+    elif [[ $# -ne 1 ]]; then
+        die "Too many arguments in '$(basename $0) kernels' command. Please use \`$0 help\` for help."
+    else 
+        KERNEL_VERSION=$1
+        if [[ "$KERNEL_VERSION" != @(5.10|5.10-no-acpi|6.1) ]]; then
+            die "Unsupported kernel version: '$KERNEL_VERSION'. Please use \`$0 help\` for help."
+        fi
+    fi
 
-build_rootfs ubuntu-22.04 jammy
-build_initramfs
+    clone_amazon_linux_repo
 
-clone_amazon_linux_repo
+    # Apply kernel patches on top of AL configuration
+    apply_kernel_patches_for_ci
 
-# Apply kernel patches on top of AL configuration
-apply_kernel_patches_for_ci
+    # Create the directory in which we will store the kernels
+    mkdir -p $OUTPUT_DIR
```

Comment:
I missed this before. I think this should move either at the beginning or in `main` since both rootfs and kernels  use it.